api/v1alpha1/nodepool_types.go

```diff
@@ -286,6 +287,19 @@ type NodePoolPlatform struct {
 
 	// IBMCloud defines IBMCloud specific settings for components
 	IBMCloud *IBMCloudPlatformSpec `json:"ibmcloud,omitempty"`
+
+	// Kubevirt specifies the configuration used when operating on KubeVirt platform.
+	//
+	// +optional
+	// +immutable
+	Kubevirt *KubevirtNodePoolPlatform `json:"kubevirt,omitempty"`
+}
+
+// KubevirtNodePoolPlatform specifies the configuration of a NodePool when operating
+// on KubeVirt platform.
+type KubevirtNodePoolPlatform struct {
+	// VirtualMachineInstance Spec contains the VirtualMachineInstance specification.
+	VirtualMachineInstanceSpec kubevirtv1.VirtualMachineInstanceSpec `json:"vmispec,omitempty"`
```

Comment:
Hm, I don't think that we can use `vmispec`. How about `template`?

Reply:
Changed